test/cpp/end2end/xds/xds_end2end_test.cc

```diff
@@ -1324,18 +1324,47 @@ class XdsEnd2endTest : public ::testing::TestWithParam<TestType> {
     });
   }
 
-  static Listener BuildListener(const RouteConfiguration& route_config) {
+  AdsServiceImpl::ResponseState RouteConfigurationResponseState(int idx) const {
+    AdsServiceImpl* ads_service = balancers_[idx]->ads_service();
+    if (GetParam().enable_rds_testing()) {
+      return ads_service->rds_response_state();
+    }
+    return ads_service->lds_response_state();
+  }
+
+  void SetListenerAndRouteConfiguration(
+      int idx, Listener listener, const RouteConfiguration& route_config) {
```

Comment:
If we want to reuse this for the server side, we would also want a boolean to let this know what kind of listener is being used. (For a server-side listener, we would want to modify the HttpConnectionManager within the default filter chain)